torch/_inductor/lowering.py

```diff
@@ -5717,6 +5714,42 @@ def resize_storage_bytes_(variable, new_size):
     return variable
 
 
+@register_lowering(aten.resize, type_promotion_kind=None)
+def resize(x, size, *, memory_format=None):
+    x.realize()
+    dtype_item_size = get_dtype_size(x.get_dtype())
+    old_numel = V.graph.sizevars.evaluate_static_shape(sympy_product(x.get_size()))
+    new_numel = V.graph.sizevars.evaluate_static_shape(sympy_product(size))
```

Comment:
Putting guards unconditionally here is not great. You should do something like `statically_known_equals`.